Align tile widths

DennisTsiang opened the pull request:
Aligned tile widths so that they are all consistent on the site, and give them a padding on either side.

Commit:
Align tile sections on homepage

Tile sections were slightly indented compared to the other sections.
This commit aligns the tiles by adding a negative margin to the section
and giving each tile 3px padding on either side.
On mobile view paddings are removed so the tiles fill the full width of
the screen.

Commit:
Refactor small tiles

Refactored small blog-tile classes so that it now extends from a new
small-tile class. This helps reduce redundancy and removes the
dependency of the class to the page it's being rendered on.

Commit:
Change product story tiles to have padding either side

A 3px padding is added to either side. This keeps it consistent with the
other tiles. Container is now 6px longer to account for this.

Commit:
Change small product story tile and homepage tile to extend from new tile

homepage-tile and product-story-small-tile are very similar and have
duplicated styling in some classes. By changing the
product-story-small-tile and homepage-tile to extend from showcase-tile
it helps to reduce redundancy. Also:
- Renamed the class and key values from "title" to "description" as I
  felt "description" was more accurate name for its purpose.

Commit:
Add padding to person tiles

Person tiles now have a 3px padding on either side to keep it consistent
with the other tiles on the site.

Commit:
Change blog tiles to have padding on either side

Keeps it consistent with rest of the site. homepage-tile and
blog-small-tile are now the same so can be completely replaced by
small-tile.

Commit:
Add some padding to other tiles

Keeps them all to 960px. Also changed tiles on tablet mode to fill full
width too. Also:
- Give role image "object-fit:cover" property which
  allows the image to fill the container and auto crop when resized
- Fix text resizing of roles tiles
- Made background colour darker to make tiles stand out more.

neilvanbeinum commented on source/people.erb:
```diff
@@ -14,41 +14,41 @@ email_domain: "unboxed.co"
-        <div class="person-tile__image-container">
-          <img class="person-tile__image" src="/assets/images/people/<%= person.short_name %>.png" alt="<%= person.name %>" />
-        </div>
+    <% data.people.each do |person| %>
```
I don't think this needs indenting

neilvanbeinum commented on source/assets/stylesheets/styles/_homepage_tile.scss:
```diff
@@ -2,12 +2,15 @@
     width: 33.33%;
+    padding-left: 3px;
+    padding-right: 3px;
+
```
Extra line ✂️  :)

DennisTsiang replied:
That file gets changed in a later commit (3c49b35)

neilvanbeinum reviewed (changes requested):
Left a couple of very minor comments. Overall I think it looks good. As we discussed in person, I think the product story tiles at tablet width might need to kept more consistent with the existing design.

Commit:
Fix object-fit styling on IE and Edge

The CSS property object-fit is not supported in older IE browsers and
Edge (though it's currently being implemented for Edge) resulting in
images being skewd when window is resized. To fix this we include a
javascript file created by bfred-it on github (link:
https://github.com/bfred-it/object-fit-images/).

As explained in his repo, by including the styling "font-family:
'object-fit: cover' " in the stylings for the image containers, we can
fix this problem.

neilvanbeinum commented on source/assets/stylesheets/styles/_small_tile.scss:
```diff
@@ -0,0 +1,81 @@
+  border: 1px solid darken($light-gray, 5%);
+  box-sizing: border-box;
+  display: block;
+  width: 100%;ht: 190px;
```
The text after the first `;` looks like a typo that can be deleted

neilvanbeinum commented on source/product-stories.erb:
```diff
@@ -89,7 +89,7 @@ product_stories:
         <div class="<%= class_name %>__content">
           <div class="<%= class_name %>__client"><%= product_story.client %></div>
-          <div class="<%= class_name %>__title"><%= product_story.title %></div>
+          <div class="<%= class_name %>__paragraph"><%= product_story.title %></div>
```
Why did we rename this to `paragraph`? I agree that calling the content `title` isn't accurate, but paragraph is a bit generic to my mind. How about `description`? Whichever name we use I suggest that the class name and the key match in the same way as we've done for `client`.

neilvanbeinum commented on source/assets/stylesheets/styles/_roles_tile.scss:
```diff
@@ -86,6 +86,7 @@
   width: 100%;
   margin: auto;
   object-fit: cover;
```
It would be good to add a comment here saying why we've got this unusual rule

DennisTsiang replied:
Good point, I'll add a comment explaining this in the commit

neilvanbeinum replied:
It's good we've got the description in the commit message but I also think a comment in the CSS file itself would be useful. A short one-liner should do it just explaining that the object fit JS library requires this rule to work.

DennisTsiang replied:
Added short comments to the object-fit property now in the last commit
